src/v2/coins/abstractUtxoCoin.js

```diff
@@ -1121,12 +1129,33 @@ class AbstractUtxoCoin extends BaseCoin {
       const baseKeyPath = deriveKeys(deriveKeys(keys, 0), 0);
       const userKeyArray = deriveKeys(baseKeyPath, 0);
       const changeKeyArray = deriveKeys(baseKeyPath, 1);
+      const userKeyArraySW = deriveKeys(baseKeyPath, 10);
+      const changeKeyArraySW = deriveKeys(baseKeyPath, 11);
 
       // Collect the unspents
       const addressesById = {};
-      const userUnspents = yield queryBlockchainUnspentsPath(userKeyArray, '/0/0/0');
-      const changeUnspents = yield queryBlockchainUnspentsPath(changeKeyArray, '/0/0/1');
-      const unspents = userUnspents.concat(changeUnspents);
+      // Gather the queries, to be executed in parallel
+      let queries = [
+        queryBlockchainUnspentsPath(userKeyArray, '/0/0/0', false),
+        queryBlockchainUnspentsPath(changeKeyArray, '/0/0/1', false)]
+      if (!params.ignoreSegwit) {
+        queries = queries.concat([queryBlockchainUnspentsPath(userKeyArraySW, '/0/0/10', true), queryBlockchainUnspentsPath(changeKeyArraySW, '/0/0/11', true)]);
+      }
+      const queryResponse = yield Promise.all(queries);
+      const userUnspents = queryResponse[0];
+      const changeUnspents = queryResponse[1];
+      let userUnspentsSW, changeUnspentsSW;
+      if (!params.ignoreSegwit) {
```

Comment:
I think this can be coalesced with the next if statement, which checks for the opposite condition.